### PR TITLE
fix: 1845 - Specify which sites require a defined axis

### LINKED ISSUE #1845: Central site selection allows user to select sites without a defined axis
**Describe the bug**
The selection of SiteA in the SDF module should only allow the user to select sites for which there is a defined axis.

**To Reproduce**
Appears in Step 9c of the Dissolve Water example

**Expected behaviour**
The site selection should only display 'COM'

**Screenshots**
image

**Operating System:**
 - OS:  OSX
 - OS version / flavour:  Ventura

### PR DESCRIPTION
**Pull request by rhinoella**
Closes #1845

Only allows user to select sites with defined axes for origin sites.

### CODE REVIEW

**Comment by rhinoella** [comment]
@trisyoungs I could not tell whether or not `ModifierOSitesModule` `SiteA` keyword requires a defined axis, could you please double-check.

**Comment by rhinoella** [comment]
- [ ] Will need to update tests as well.

File: src/modules/histogramCN/histogramCN.cpp [hunk 1]
```diff
@@ -18,7 +18,8 @@ HistogramCNModule::HistogramCNModule() : Module(ModuleTypes::HistogramCN)
     keywords_.addTarget<ConfigurationKeyword>("Configuration", "Set target configuration for the module", targetConfiguration_);
 
     keywords_.setOrganisation("Options", "Sites", "Specify sites defining the interaction A...B.");
-    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the reference origin", a_);
+    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the reference origin", a_,
+                                            true);
```
Comment (trisyoungs) [inline src/modules/histogramCN/histogramCN.cpp]: Don't need axes for this one.
```suggestion
    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the reference origin", a_);
```

File: src/modules/intraDistance/intraDistance.cpp [hunk 2]
```diff
@@ -17,7 +17,8 @@ IntraDistanceModule::IntraDistanceModule() : Module(ModuleTypes::IntraDistance)
     keywords_.addTarget<ConfigurationKeyword>("Configuration", "Set target configuration for the module", targetConfiguration_);
 
     keywords_.setOrganisation("Options", "Sites", "Specify sites defining the distance interaction A-B.");
-    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_);
+    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_,
+                                            true);
```
Comment (trisyoungs) [inline src/modules/intraDistance/intraDistance.cpp]: Or this one.
```suggestion
    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_);
```

File: src/modules/siteRDF/siteRDF.cpp [hunk 3]
```diff
@@ -21,7 +21,8 @@ SiteRDFModule::SiteRDFModule() : Module(ModuleTypes::SiteRDF)
     keywords_.addTarget<ConfigurationKeyword>("Configuration", "Set target configuration for the module", targetConfiguration_);
 
     keywords_.setOrganisation("Options", "Sites");
-    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_);
+    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_,
+                                            true);
```
Comment (trisyoungs) [inline src/modules/siteRDF/siteRDF.cpp]:
```suggestion
    keywords_.add<SpeciesSiteVectorKeyword>("SiteA", "Set the site(s) 'A' which are to represent the origin of the RDF", a_);
```

**Review by trisyoungs — APPROVED** [review]
Good catch. Some detail that obviously got lost in the analysis refactor.

Couple of reversions, then good to go.

**Comment by trisyoungs** [comment]
> @trisyoungs I could not tell whether or not `ModifierOSitesModule` `SiteA` keyword requires a defined axis, could you please double-check.
No axes required for that module.  Failing tests are probably caused by requiring axes on sites that didn't need them (see my review comments).

**Pull request merged**